code/modules/mod/mod_paint.dm

```diff
@@ -148,20 +148,27 @@
 	. = ..()
 	name = "MOD [skin] skin applier"
 
-/obj/item/mod/skin_applier/pre_attack(atom/attacked_atom, mob/living/user, params)
+/obj/item/mod/skin_applier/interact_with_atom(atom/attacked_atom, mob/living/user, params)
 	if(!istype(attacked_atom, /obj/item/mod/control))
-		return ..()
+		return NONE
 	var/obj/item/mod/control/mod = attacked_atom
 	if(mod.active || mod.activating)
 		balloon_alert(user, "suit is active!")
-		return TRUE
+		return ITEM_INTERACT_BLOCKING
 	if(!(skin in mod.theme.variants))
 		balloon_alert(user, "incompatible theme!")
-		return TRUE
+		return ITEM_INTERACT_BLOCKING
 	mod.theme.set_skin(mod, skin)
 	balloon_alert(user, "skin applied")
 	qdel(src)
-	return TRUE
+	return ITEM_INTERACT_SUCCESS
 
 /obj/item/mod/skin_applier/honkerative
+	desc = /obj/item/mod/skin_applier::desc + " Additionally, this kit attempts to install a waddling module. Honk!"
 	skin = "honkerative"
+
+/obj/item/mod/skin_applier/honkerative/interact_with_atom(obj/item/mod/control/controlunit, mob/living/user, params)
+	. = ..()
+	if(. != ITEM_INTERACT_SUCCESS)
```

Comment:
```suggestion
	if(!(. & ITEM_INTERACT_SUCCESS))
```
i know in this context it doesnt matter much, but just for explicitness' sake since we're dealing with bitflags